src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -413,6 +413,7 @@ public static List<BranchPoint> GetBranchPoints(MethodDefinition methodDefinitio
             bool isAsyncStateMachineMoveNext = IsMoveNextInsideAsyncStateMachine(methodDefinition);
             bool isMoveNextInsideAsyncStateMachineProlog = isAsyncStateMachineMoveNext && IsMoveNextInsideAsyncStateMachineProlog(methodDefinition);
             bool skipFirstBranch = IsMoveNextInsideEnumerator(methodDefinition);
+            bool skipSecondBranch = skipFirstBranch && instructions.All(i => i.OpCode.Code != Code.Switch);
```

Comment:
This rule seem too generic, if inside method we have also switch instruction we won't skip that branch
```cs
        public System.Collections.Generic.IEnumerable<int> One()
        {
            int result = 0;
            switch (new Random().Next())
            {
                case 1:
                    {
                        result = 23;
                        break;
                    }
                case 2:
                    {
                        result = -5;
                        break;
                    }
                case 4:
                    {
                        result = -3;
                        break;
                    }
                default:
                    break;
            }

            if (result < 0)
            {
                throw new Exception();
            }

            yield return 1;
        }
```
Branch on `10` is wrong  `12` is `switch` and `33` the `if`
![image](https://user-images.githubusercontent.com/7894084/79591099-583e7180-80d8-11ea-8e37-7668c8c01eff.png)

Reply:
Darn, that's true. I added a failing test case for this scenario. Currently, I have no idea how to solve this. Any help would be much appreciated.

Reply:
Sheesh, for some reason that test only fails in debug, but not in release configuration. When I use LINQPad to compare the generated MSIL in both cases it has the same number of branches though...

Reply:
Need some time to understand how to fix this edge case, not simple...seem that we need to understand that we are inside enumerator and follow jump of first condition and skip that similar to:
```
if/switch (trim)

   another if/switch

case/else (trim)

   another if/switch

case/else (trim)

```
Not simple...this is the hardest thing with coverage 😄

Reply:
I don't think following the execution flow is necessary, because it will always be either the first or the first and the second branch we need to trim/skip.

I just pushed a commit the fixes this issue. However, I'd feel uncomfortable using it until I understand why the test case didn't fail in release builds.

Reply:
Have you compared with ILSpy?

Reply:
Yes, of course. Did I miss something?

Reply:
Oh you mean this in regard to the different integration test result of debug vs release. I couldn't find any structural differences.

<details><summary>debug MSIL</summary>

.method private final hidebysig newslot virtual 
	instance bool MoveNext () cil managed 
{
	.override method instance bool [System.Runtime]System.Collections.IEnumerator::MoveNext()
	// Method begins at RVA 0x1012c
	// Code size 142 (0x8e)
	.maxstack 2
	.locals init (
		[0] int32,
		[1] int32,
		[2] int32
	)

	IL_0000: ldarg.0
	IL_0001: ldfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_0006: stloc.0
	IL_0007: ldloc.0
	IL_0008: brfalse.s IL_0012

	IL_000a: br.s IL_000c

	IL_000c: ldloc.0
	IL_000d: ldc.i4.1
	IL_000e: beq.s IL_0014

	IL_0010: br.s IL_0016

	IL_0012: br.s IL_0018

	IL_0014: br.s IL_0085

	IL_0016: ldc.i4.0
	IL_0017: ret

	IL_0018: ldarg.0
	IL_0019: ldc.i4.m1
	IL_001a: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_001f: nop
	IL_0020: ldarg.0
	IL_0021: ldarg.0
	IL_0022: ldfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::n
	IL_0027: stloc.1
	IL_0028: ldloc.1
	IL_0029: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>s__2'
	IL_002e: ldarg.0
	IL_002f: ldfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>s__2'
	IL_0034: stloc.2
	IL_0035: ldloc.2
	IL_0036: switch (IL_0049, IL_0053, IL_005d)

	IL_0047: br.s IL_0067

	IL_0049: ldarg.0
	IL_004a: ldc.i4.s 10
	IL_004c: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<result>5__1'
	IL_0051: br.s IL_0070

	IL_0053: ldarg.0
	IL_0054: ldc.i4.s 11
	IL_0056: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<result>5__1'
	IL_005b: br.s IL_0070

	IL_005d: ldarg.0
	IL_005e: ldc.i4.s 12
	IL_0060: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<result>5__1'
	IL_0065: br.s IL_0070

	IL_0067: ldarg.0
	IL_0068: ldc.i4.m1
	IL_0069: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<result>5__1'
	IL_006e: br.s IL_0070

	IL_0070: ldarg.0
	IL_0071: ldarg.0
	IL_0072: ldfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<result>5__1'
	IL_0077: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>2__current'
	IL_007c: ldarg.0
	IL_007d: ldc.i4.1
	IL_007e: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_0083: ldc.i4.1
	IL_0084: ret

	IL_0085: ldarg.0
	IL_0086: ldc.i4.m1
	IL_0087: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_008c: ldc.i4.0
	IL_008d: ret
} // end of method '<OneWithSwitch>d__2'::MoveNext

</details>

<details><summary>release MSIL</summary>

.method private final hidebysig newslot virtual 
	instance bool MoveNext () cil managed 
{
	.override method instance bool [System.Runtime]System.Collections.IEnumerator::MoveNext()
	// Method begins at RVA 0xeca4
	// Code size 92 (0x5c)
	.maxstack 2
	.locals init (
		[0] int32,
		[1] int32 result,
		[2] int32
	)

	IL_0000: ldarg.0
	IL_0001: ldfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_0006: stloc.0
	IL_0007: ldloc.0
	IL_0008: brfalse.s IL_0010

	IL_000a: ldloc.0
	IL_000b: ldc.i4.1
	IL_000c: beq.s IL_0053

	IL_000e: ldc.i4.0
	IL_000f: ret

	IL_0010: ldarg.0
	IL_0011: ldc.i4.m1
	IL_0012: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_0017: ldarg.0
	IL_0018: ldfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::n
	IL_001d: stloc.2
	IL_001e: ldloc.2
	IL_001f: switch (IL_0032, IL_0037, IL_003c)

	IL_0030: br.s IL_0041

	IL_0032: ldc.i4.s 10
	IL_0034: stloc.1
	IL_0035: br.s IL_0043

	IL_0037: ldc.i4.s 11
	IL_0039: stloc.1
	IL_003a: br.s IL_0043

	IL_003c: ldc.i4.s 12
	IL_003e: stloc.1
	IL_003f: br.s IL_0043

	IL_0041: ldc.i4.m1
	IL_0042: stloc.1

	IL_0043: ldarg.0
	IL_0044: ldloc.1
	IL_0045: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>2__current'
	IL_004a: ldarg.0
	IL_004b: ldc.i4.1
	IL_004c: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_0051: ldc.i4.1
	IL_0052: ret

	IL_0053: ldarg.0
	IL_0054: ldc.i4.m1
	IL_0055: stfld int32 Coverlet.Core.Samples.Tests.Yield/'<OneWithSwitch>d__2'::'<>1__state'
	IL_005a: ldc.i4.0
	IL_005b: ret
} // end of method '<OneWithSwitch>d__2'::MoveNext

</details>